bin/create_staging_db

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash -e
+#
+# Creates an empty MARTI database for staging and grants the specified user
+# all privileges in the new database.
+#
+
+if [[ $# -gt 1 ]]; then
+  echo "usage: $0 [db_user]"
+  echo
+  echo "  db_user: The database user who will be granted full privileges to"
+  echo "           the staging database (default: marti_dice@localhost)."
+  exit 1
+fi
+
+readonly db_user=${1:-marti_dice@localhost}
+
+read -p "Enter password for database user 'root': " -s db_password
+echo
+readonly db_defaults_file=$(mktemp)
+chmod 600 $db_defaults_file
+cat > $db_defaults_file <<EOF
```

Comment:
Would't it be clearer if we just use `echo "" > filename` to create a new file?

Reply:
Point taken.  I had _here documents_ on my mind. :smile:

I tried the following equivalent `echo` command:

```bash
echo "[client]
password=$db_password" > $db_defaults_file
```

While it's shorter, I'm not sure it reads better because a) the first line is not in the same column as the remaining lines, and b) the target file is at the end of the command in some arbitrary column.  I can address (a) by dropping a new line and escaping the newline character:

```bash
echo "\
[client]
password=$db_password" > $db_defaults_file
```

And, similarly, (b) can be partially addressed by dropping the output redirection to a new line:

```bash
echo "\
[client]
password=$db_password" \
  > $db_defaults_file
```

But now we're back at the same line count, and I would probably still argue the here document syntax is a bit more readable.

The top five results of a search for "bash write multiline file" all seem to suggest here documents as being idiomatic, so I think I'll just stick with the current "cat script" usage (which is apparently how The Linux Documentation Project [refers](http://tldp.org/LDP/abs/html/here-docs.html) to them in Example 19-3).